examples/models/VGG.cpp

```diff
@@ -0,0 +1,631 @@
+    auto softmax_prim = softmax_forward(softmax_pd);
+
+    net.push_back(softmax_prim);
+    net_args.push_back(
+            {{DNNL_ARG_SRC, ip3_dst_mem}, {DNNL_ARG_DST, ip3_dst_mem}});
+    /* ------------------------------------------ */
+    // Execute model
+
+    for (size_t i = 0; i < net.size(); ++i) {
+        net[i].execute(s, net_args[i]);
+        // auto dst_mem = net_args.at(i).at(DNNL_ARG_DST);
+        // std::vector<float> temp(dst_mem.get_desc().get_size() /
+        // sizeof(float)); read_from_dnnl_memory(temp.data(), dst_mem); auto
+        // kind = net.at(i).get_kind(); std::string kind_str; switch (kind) {
+        //     case dnnl::primitive::kind::convolution:
+        //         kind_str = "Convolution: ";
+        //         break;
+        //     case dnnl::primitive::kind::eltwise: kind_str = "Eltwise: ";
+        //     break; case dnnl::primitive::kind::pooling: kind_str = "Pooling:
+        //     "; break; case dnnl::primitive::kind::batch_normalization:
+        //         kind_str = "Bnorm: ";
+        //         break;
+        //     case dnnl::primitive::kind::inner_product:
+        //         kind_str = "Inner Product: ";
+        //         break;
+        // }
+        // std::cout << kind_str << std::endl;
+        // hash_and_print(temp, true, 10);
+    }
+
+    // Wait for completion.
+    s.wait();
+    std::vector<float> result(product(ip3_dst_mem.get_desc().dims()));
+    read_from_dnnl_memory(result.data(), ip3_dst_mem);
+    for (size_t i = 0; i < BATCH; i++) {
+        auto index = std::max_element(
+                result.begin() + i * 1000, result.begin() + (i + 1) * 1000);
+        //std::cout << "classed as "
+        //          << std::distance(result.begin() + i * 1000, index)
+        //          << ", value " << (index != std::end(result) ? *index : 0.f)
+        //          << std::endl;
+    }
+
+#undef STAMP_OUT_WEIGHTS_CONV
+}
+
+void run_vgg(engine &eng, stream &s, std::vector<primitive> &net,
+        std::vector<std::unordered_map<int, memory>> net_args,
+        std::vector<char> &input_image, int times) {
+    // Execute model
+    for (int j = 0; j < times; ++j) {
+        for (size_t i = 0; i < net.size(); ++i) {
+            net[i].execute(s, net_args[i]);
+        }
+    }
+    // Wait for completion.
+    s.wait();
+}
+
+void check_vgg(engine &eng, stream &s, std::vector<primitive> &net,
+        std::vector<std::unordered_map<int, memory>> net_args,
+        std::vector<char> &input_image) {
+    run_vgg(eng, s, net, net_args, input_image, 1);
+}
+
+void do_it(engine::kind engine_kind) {
+    engine eng(engine_kind, 0);
+    stream s(eng);
+
+    std::vector<primitive> net;
+    std::vector<std::unordered_map<int, memory>> net_args;
+
+    std::vector<char> input_image;
+
+    std::cout << "Setting up... for BATCH SIZE = " << BATCH << "\n";
+    set_up_vgg(eng, s, net, net_args, input_image);
+    //     int times = 1;
```

Comment:
Is it intentional to not do any warmup? Will the measured performance be fine if no warmup is done?

Reply:
We did do the warmup before the performance numbers, this code was commented by mistake. Reverted it back now.